File: 0590_N-ary Tree Postorder Traversal/N-aryTreePostorderTraversal.java
class Solution {
    public List<Integer> postorder(Node root) {
        LinkedList<Integer> list = new LinkedList<>();
        if(root == null) return list;
        Stack<Node> stack = new Stack<>();
        stack.push(root);
        while(!stack.isEmpty()){
            Node n = stack.pop();
            if(n.children != null){
                for(Node node: n.children){
                    stack.push(node);
                }
            }
            list.addFirst(n.val);
        }
        
        return list;
    }
}

File: 0590_N-ary Tree Postorder Traversal/N-aryTreePostorderTraversal.cs
public class Solution {
    public IList<int> Postorder(Node root) {
        var ret = new List<int>();
        Postorder(root, ret);
        return ret;
    }
    
    private void Postorder(Node node, IList<int> list)
    {
        if(node == null) return;
        if(node.children!=null){
            foreach(var n in node.children){
                Postorder(n,list);
            }
        }
        
        list.Add(node.val);
        
    }
}